apps/server/src/modules/files/repo/files.repo.spec.ts

```diff
@@ -0,0 +1,523 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+
+import { MongoMemoryDatabaseModule } from '@shared/infra/database';
+import { StorageProviderEntity } from '@shared/domain';
+import { storageProviderFactory } from '@shared/testing';
+
+import { FileOwnerModel, FilePermissionReferenceModel } from '../domain';
+import { FileEntity, FilePermissionEntity } from '../entity';
+import { userFileFactory } from '../entity/testing';
+import { FilesRepo } from './files.repo';
+
+describe(FilesRepo.name, () => {
+	let repo: FilesRepo;
+	let em: EntityManager;
+	let module: TestingModule;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			imports: [
+				MongoMemoryDatabaseModule.forRoot({
+					entities: [FileEntity, StorageProviderEntity],
+				}),
+			],
+			providers: [FilesRepo],
+		}).compile();
+
+		repo = module.get(FilesRepo);
+		em = module.get(EntityManager);
+	});
+
+	beforeEach(async () => {
+		await em.nativeDelete(FileEntity, {});
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('defined', () => {
+		it('repo should be defined', () => {
+			expect(repo).toBeDefined();
+		});
+
+		it('entity manager should be defined', () => {
+			expect(em).toBeDefined();
+		});
+
+		it('should implement entityName getter', () => {
+			expect(repo.entityName).toBe(FileEntity);
+		});
+	});
+
+	describe('findForCleanup', () => {
+		it('should return files marked for deletion according to given params', async () => {
+			const file: FileEntity = userFileFactory.build({ deletedAt: new Date() });
+
+			await em.persistAndFlush(file);
+			em.clear();
+
+			const thresholdDate = new Date();
+
+			const result = await repo.findForCleanup(thresholdDate, 3, 0);
+
+			expect(result.length).toEqual(1);
+			expect(result[0].id).toEqual(file.id);
+		});
+
+		it('should not return files which are not marked for deletion', async () => {
+			const file = userFileFactory.build({ deletedAt: undefined });
+
+			await em.persistAndFlush(file);
+			em.clear();
+
+			const thresholdDate = new Date();
+			const result = await repo.findForCleanup(thresholdDate, 3, 0);
+
+			expect(result.length).toEqual(0);
+		});
+
+		it('should not return files where deletedAt is after threshold', async () => {
+			const thresholdDate = new Date();
+			const file = userFileFactory.build({ deletedAt: new Date(thresholdDate.getTime() + 10) });
+
+			await em.persistAndFlush(file);
+			em.clear();
+
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			expect(file.deletedAt!.getTime()).toBeGreaterThan(thresholdDate.getTime());
+
+			const result = await repo.findForCleanup(thresholdDate, 3, 0);
+
+			expect(result.length).toEqual(0);
+		});
+	});
+
+	const storageProvider = storageProviderFactory.buildWithId();
```

Comment:
those constants should be declared before tests I think

Reply:
Correct, thanks, forgot we use the AAA-pattern in this project.